src/CondenserDotNet.Server/RoutingHost.cs

```diff
@@ -21,13 +21,16 @@ public class RoutingHost
         private readonly ILogger<RoutingHost> _logger;
         private readonly RoutingData _routingData;
         private readonly Func<IConsulService> _serviceFactory;
+        private readonly Func<ICurrentState> _statsFactory;
         private string _lastConsulIndex = string.Empty;
 
         public RoutingHost(CustomRouter router, CondenserConfiguration config, ILoggerFactory logger,
-            RoutingData routingData, IEnumerable<IService> customRoutes, Func<IConsulService> serviceFactory)
+            RoutingData routingData, IEnumerable<IService> customRoutes, Func<IConsulService> serviceFactory, 
+            Func<ICurrentState> statsFactory)
```

Comment:
We can remove the stats factory here if we switch it to DI injected on the service